### PR TITLE
feat: add fish

### PR DESCRIPTION
**Pull request by etcusrvar**
_Motivation:_ Add fish shell
_Linked issues:_ #1757

https://fishshell.com

This work was started but left incomplete several years ago in #1757.
I used that as a starting point.
This installs the current fish release, 3.1.2.
Installed and running successfully on avoton. 

### Checklist
- [x] Build rule `all-supported` completed successfully
  * compile failed on 88f6281; added $(ARMv5_ARCHS) to UNSUPPORTED_ARCHS
  * compile failed on qoriq; added $(PPC_ARCHS) to UNSUPPORTED_ARCHS
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### CODE REVIEW

**Comment by hgy59** [comment]
@etcusrvar thanks for contributing this. 
I tried to update fish-shell on #1757 but failed to rebase to current head@master (as it is too old).

File: cross/fish/Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+HOMEPAGE = https://fishshell.com
+COMMENT = fish is a smart and user-friendly command line shell for Linux, macOS, and the rest of the family.
+LICENSE = Custom
+
+UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(PPC_ARCHS)
+
+# fish needs pcre2-32
+ENV += PCRE2_CLI_FULL=1
+
```
Comment (hgy59) [inline cross/fish/Makefile:19]: with updates of 628c0bbf53b4af650cd599d4d0c24bc456f0bb21 this is not required anymore, as pcre2-32 is always included in cross/pcre2
ENV += PCRE2_CLI_FULL=1 did not work (I had to export PCRE2_CLI_FULL=1)

File: cross/fish/Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+PKG_NAME = fish
+PKG_VERS = 3.1.2
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/fish-shell/fish-shell/releases/download/$(PKG_VERS)
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/ncurses cross/pcre2
+OPTIONAL_DEPENDS = cross/bzip2 cross/readline cross/zlib
+
```
Comment (hgy59) [inline cross/fish/Makefile:10]: please remove this here, it is defined in cross/pcre2

File: cross/fish/PLIST [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+bin:bin/fish_indent
+bin:bin/fish_key_reader
+lib:lib/libpcre2-32.so.0.10.0
+lnk:lib/libpcre2-32.so
+lnk:lib/libpcre2-32.so.0
+rsc:etc/fish
+rsc:share/doc
+rsc:share/fish
+rsc:share/locale
+rsc:share/man
```
Comment (hgy59) [inline cross/fish/PLIST:11]: as the `man` executable is not available on the diskstations, all the synocommunity packages avoid to install man files.

may be the `share/doc` folder can be removed too.

File: cross/fish/PLIST [hunk 4]
```diff
@@ -0,0 +1,12 @@
+bin:bin/fish
+bin:bin/fish_indent
+bin:bin/fish_key_reader
+lib:lib/libpcre2-32.so.0.10.0
+lnk:lib/libpcre2-32.so
+lnk:lib/libpcre2-32.so.0
```
Comment (hgy59) [inline cross/fish/PLIST:6]: please remove this here. It is provided in cross/pcre2/PLIST

**Review by hgy59 — CHANGES_REQUESTED** [review]

**Comment by etcusrvar** [comment]
@hgy59, I've made the requested changes locally. If I push them here, this PR will have several checks that fail because the compiles fail, because #4365 is not merged yet.
I verified that with rebasing this PR on top of #4365, and making the requested changes, the build works and installs and runs fine.
How would you suggest I proceed?

File: cross/fish/PLIST [hunk 5]
```diff
@@ -0,0 +1,15 @@
+bin:bin/fish
+bin:bin/fish_indent
+bin:bin/fish_key_reader
+rsc:etc/fish/
```
Comment (hgy59) [inline cross/fish/PLIST:4]: I suppose that custom configurations are made in the files located in the `etc/fish` folder.
On a package update/upgrade this folder will be lost, so I propose to move these files to `${SYNOPKG_PKGDEST}/var/`.
So you must add a custom post installation to move the files to the new location and need to configure fish-shell to use this specific folder.
 
You can test whether the `Package upgrade completed successfully`  when you manually install the same package again (without prior uninstall). If your custom settings are restored then you managed it.

**Review by hgy59 — CHANGES_REQUESTED** [review]
LGTM. Just some work needed to improve the package upgrade.

File: cross/fish/PLIST (hunk 5 above)
Comment (etcusrvar) [reply cross/fish/PLIST:4]: There's several other projects that include `rsc:etc` in their cross/project/PLIST files.
For the upgrade scenario, I followed rsnapshot ([source](https://github.com/SynoCommunity/spksrc/blob/b883d5616639685198522ba48fdc672c9982c881/spk/rsnapshot/src/service-setup.sh#L9-L19)) and others as examples.
I did an install, edited /usr/local/fish/etc/* files on the system, changed the SPK_REV, upgraded, and fish/etc was successfully backed up and restored and working as expected on the system, using the restored etc files.

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

File: cross/fish/PLIST [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+rsc:share/fish/__fish_build_paths.fish
+rsc:share/fish/completions/
+rsc:share/fish/config.fish
+rsc:share/fish/functions/
+rsc:share/fish/tools/
+rsc:share/fish/vendor_completions.d/
+rsc:share/fish/vendor_conf.d/
+rsc:share/fish/vendor_functions.d/
+rsc:share/locale/
+rsc:share/pkgconfig/
```
Comment (hgy59) [inline cross/fish/PLIST:15]: sorry, didn't catch this in the first review:

pkgconfig folder should not be installed as it is used at build time only (for other packages that link to this)

**Review by hgy59 — APPROVED** [review]

**Pull request merged**